trie/verkle.go

```diff
@@ -336,27 +336,16 @@ func ProveAndSerialize(pretrie, posttrie *VerkleTrie, keys [][]byte, resolver ve
 	return p, kvps, nil
 }
 
-type set = map[string]struct{}
-
-func addKey(s set, key []byte) {
-	s[string(key)] = struct{}{}
-}
-
-func DeserializeAndVerifyVerkleProof(vp *verkle.VerkleProof, root []byte, statediff verkle.StateDiff) error {
-	rootC := new(verkle.Point)
-	rootC.SetBytes(root)
-
-	var others set = set{} // Mark when an "other" stem has been seen
+func DeserializeAndVerifyVerkleProof(vp *verkle.VerkleProof, preStateRoot []byte, postStateRoot []byte, statediff verkle.StateDiff) error {
+	// TODO: check that `OtherStems` have expected length and values.
```

Comment:
This is something that we should do. This prob needs to be done in `PreStateTreeFromProof(...)`. The pre-state tree reconstruction should consume _exactly_ all elements in `OtherStems`. If some extra value isn't used, then the proof should be considered incorrect (IMHO).

This is unrelated with all this post-state change; just something I realized that _I believe_ was the original idea of this deleted code in L339-349 & L356-359 (all that code wasn't doing anything).

To do this extra check, we don't need that code, so I'm suggesting here to be removed (since it's currently not doing anything).

Reply:
sure yes we should add this, in another PR if you prefer.